src/test/kotlin/app/services/impl/HttpKeyServiceTest.kt

```diff
@@ -0,0 +1,156 @@
+package app.services.impl
+
+import app.batch.S3Reader
+import app.configuration.HttpClientProvider
+import app.configuration.S3Configuration
+import app.exceptions.DataKeyDecryptionException
+import app.exceptions.DataKeyServiceUnavailableException
+import org.apache.hadoop.hbase.client.Connection
+import org.apache.http.HttpEntity
+import org.apache.http.StatusLine
+import org.apache.http.client.methods.CloseableHttpResponse
+import org.apache.http.client.methods.HttpPost
+import org.apache.http.impl.client.CloseableHttpClient
+import org.junit.Assert
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.ArgumentMatchers
+import org.mockito.BDDMockito.given
+import org.mockito.Mockito.*
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.boot.test.mock.mockito.MockBean
+import org.springframework.test.context.ActiveProfiles
+import org.springframework.test.context.TestPropertySource
+import org.springframework.test.context.junit4.SpringRunner
+import java.io.ByteArrayInputStream
+
+@RunWith(SpringRunner::class)
+@SpringBootTest
+@TestPropertySource(properties = [
+    "hbase.zookeeper.quorum=phoney",
+    "data.key.service.url=phoney"
+])
+class HttpKeyServiceTest {
+
+    @Before
+    fun init() {
+        this.keyService.clearCache()
+        reset(this.httpClientProvider)
+    }
+
+    @Test
+    fun testDecryptKeyOk() {
+        val responseBody = """
```

Comment:
Can we please have test names like "decryptKey_will_call_dks_and_return_data_when_input_is_valid" or something simmilar?

Reply:
As a sample see what we now have in HTME